### PR TITLE
Functions to get informations of ALFASIM reports:

### PR DESCRIPTION
**Pull request by Gustavo-Dagostini**
-variables_names
-number_of_variable
-number_of_probes
-get_probe_location
-get_number_of_edges
-get_edge_names

Absolute Pressure (Conn 1 (8,5 [m])) [bar]
This represents a variable measurement. 
The first part, before the parentheses, is the name of the measured variable, which is 'Absolute Pressure.' 
After the parentheses, the information in '(Conn 1 (8,5 [m])' indicates:

'Conn 1' refers to the edge "Conn 1".
'8.5 [m]' specifies the position of the probes at 8.5 meters from some reference point.
'[bar]' indicates the unit of measurement, which is in bars."

So in the determination of reading logic were created "variables" that position these specific strings, and then do we use de strip function between th positions.
**It is important do not forget replace comma to dot.

### CODE REVIEW

File: src/xls_reader/xls_reader.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                for item3 in units:
+                    if (item2 == "Absolute Pressure" and item3 != 'bar') or (item2 == "Holdup" and item3 != 'm3/m3') or (item2 == "Time" and item3 != 's'):
+                        continue
+                    else:
+                        dado = f"{item2} ({item1} ({item_} [m])) [{item3}]"
+                        aux_2[item2] = dataframe[dado].tolist()
+                        aux["variables"].update(aux_2)
+        variables_dictionaries.append(aux)
+
+    return variables_dictionaries
```
Comment (rafacerq) [inline src/xls_reader/xls_reader.py:98]: For this function, I thought in the following solution:
```

def create_results_dictionary(dataframe):
    probe_positions = get_probe_position(dataframe)
    variables_names = get_variables_names(dataframe)
    edge_names = get_edge_names(dataframe)
    units = get_variable_units(dataframe)

    results_dict = {}
    for edge_name in edge_names:
        if edge_name not in results_dict:
            results_dict[edge_name] = {}
        for probe_position in probe_positions:
            if f'{probe_position:g}' not in results_dict[edge_name]:
                results_dict[edge_name][f'{probe_position:g}'] = {'position': probe_position}
            for variables_name in variables_names:
                for unit in units:
                    probe_position_string = f'{probe_position:g}'.replace('.', ',')
                    column_name = f'{variables_name} ({edge_name} ({probe_position_string} [m])) [{unit}]'
                    if column_name in dataframe.keys():
                        results_dict[edge_name][f'{probe_position:g}'][variables_name] = {'unit': unit,
                                                                                          'value': np.array([],
                                                                                                            dtype=float)
                                                                                          }

    return results_dict
```
Note that I modified the function name.
It's easier to understand. Please check it out.

File: src/xls_reader/xls_reader.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+def get_variables_names(dataframe):
+    variables_names = ['Time']
+    column_names = dataframe.keys()
+    for item in column_names[1:]:
+        first_open_parentheses = item.find("(")
+        variables_names.append(item[:first_open_parentheses-1].strip())
+    variables_single_names = set(variables_names)
+    return sorted(variables_single_names)
```
[the remark this answers is not in the document]
Comment (Gustavo-Dagostini) [reply src/xls_reader/xls_reader.py:8]: Every similar occurrences were modified and are up to date

File: src/xls_reader/xls_reader.py (hunk 1 above)
Comment (Gustavo-Dagostini) [reply src/xls_reader/xls_reader.py]: Ok. The strategy I used was a little be different so I could not add the units key and needed to create a list with dicts. The difference is that your solution attribute to the "key name" a probe position and in mine "probe position" was the "key name", cause of that was missing a Key in my solution. But it is alreadysolved and up to date.

File: src/xls_reader/xls_reader.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+def get_number_of_variables(dataframe):
+    variables_names = ['Time']
+    column_names = dataframe.keys()
```
[the remark this answers is not in the document]
Comment (Gustavo-Dagostini) [reply src/xls_reader/xls_reader.py:16]: Ok. Added

File: src/xls_reader/xls_reader.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        first_close_brackets = item.find("]")
+        second_close_brackets = item.find("]", first_close_brackets + 1)
+        units_names.append(item[second_open_brackets+1:second_close_brackets].strip())
+    single_units_names = set(units_names)
+    return sorted(single_units_names)
+
+def create_variables_dictionaries(dataframe):
+    variables_dictionaries = []
+    probe_positions = get_probe_position(dataframe)
+    variables_names = get_variables_names(dataframe)
```
[the remark this answers is not in the document]
Comment (Gustavo-Dagostini) [reply src/xls_reader/xls_reader.py:89]: Ok. Changed.

**Review by rafacerq — APPROVED** [review]

**Pull request merged**